contracts/bots/SimpleCompoundErc20FLIArb.sol

```diff
@@ -0,0 +1,547 @@
+/*
+    Copyright 2021 Set Labs Inc.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+    SPDX-License-Identifier: Apache License, Version 2.0
+*/
+pragma solidity 0.6.10;
+pragma experimental ABIEncoderV2;
+
+import { SafeCast } from "@openzeppelin/contracts/utils/SafeCast.sol";
+import { SafeMath } from "@openzeppelin/contracts/math/SafeMath.sol";
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { IUniswapV2Factory } from "@uniswap/v2-core/contracts/interfaces/IUniswapV2Factory.sol";
+
+import { Account, Actions, ISoloMargin } from "../interfaces/ISoloMargin.sol";
+import { DydxFlashloanBase } from "../lib/DydxFlashloanBase.sol";
+import { ICallee } from "../interfaces/ICallee.sol";
+import { ICErc20 } from "../interfaces/ICErc20.sol";
+import { ICompoundLeverageModule } from "../interfaces/ICompoundLeverageModule.sol";
+import { IDebtIssuanceModule } from "../interfaces/IDebtIssuanceModule.sol";
+import { ISetToken } from "../interfaces/ISetToken.sol";
+import { IUniswapV2Router } from "../interfaces/IUniswapV2Router.sol";
+import { IWETH } from "../interfaces/IWETH.sol";
+import { PreciseUnitMath } from "../lib/PreciseUnitMath.sol";
+
+/**
+ * @title SimpleCompoundErc20FLIArb
+ * @author Set Protocol
+ *
+ * Smart contract that performs creation redemption arbitrage on NAV of a FLI product against market price on Uniswap V2 or Sushiswap.
+ * Note: This contract only is compatible with FLI with 1 cToken position as collateral and 1 debt position. This contract only works
+ * with cERC20s and not cETH. Additionally, this contract is not compatible if WETH is the debt asset or if WETH is paired in the Uniswap pool
```

Comment:
What's the additional complexity here? Feels like it's a decent bet we may need to build an arb bot that is compatible with WETH and cETH at some point. Is that going to be easy to do?

Reply:
This info may be a bit better in a table. Just what is the asset (collateral, debt, pool) and what's excluded from being one of those assets